scripts/kind-e2e/e2e.sh

```diff
@@ -117,34 +147,95 @@ function test_connection() {
     netshoot_pod=$(kubectl get pods -l app=netshoot | awk 'FNR == 2 {print $1}')
 
     echo "Testing connectivity between clusters - $netshoot_pod cluster2 --> $nginx_svc_ip_cluster3 nginx service cluster3"
-    for i in 1 2; do
-        status_code=$(kubectl exec -it ${netshoot_pod} -- curl -m 30 -s -o /dev/null -w "%{http_code}" ${nginx_svc_ip_cluster3})
-        if [[ $status_code -eq 200 ]]; then
-           echo "Connection success! Status code: $status_code"
+
+    attempt_counter=0
+    max_attempts=5
+    until $(kubectl exec -it ${netshoot_pod} -- curl --output /dev/null --silent --head --fail ${nginx_svc_ip_cluster3}); do
+        if [ ${attempt_counter} -eq ${max_attempts} ];then
+          echo "Max attempts reached, connection failed!"
+          exit 1
         fi
+
+        attempt_counter=$(($attempt_counter+1))
+        sleep 10
     done
+    echo "Connection test was successful!"
+}
+
+function update_subm_pods() {
+    echo Removing submariner engine pods...
+    kubectl get pods -n submariner -l app=submariner-engine | awk 'FNR > 1 {print $1}' | xargs kubectl delete pods -n submariner
+    kubectl wait --for=condition=Ready pods -l app=submariner-engine -n submariner --timeout=60s
+    echo Removing submariner route agent pods...
+    kubectl get pods -n submariner -l app=submariner-routeagent | awk 'FNR > 1 {print $1}' | xargs kubectl delete pods -n submariner
+    kubectl wait --for=condition=Ready pods -l app=submariner-routeagent -n submariner --timeout=60s
+}
+
+function enable_logging() {
+    kubectl config use-context cluster1
+    if kubectl rollout status deploy/kibana > /dev/null 2>&1; then
+        echo Elasticsearch stack already installed, skipping...
+    else
+        echo Installing Elasticsearch...
+        es_ip=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' cluster1-control-plane | head -n 1)
+        kubectl apply -f ./kind-e2e/logging/elasticsearch.yaml
+        kubectl apply -f ./kind-e2e/logging/filebeat.yaml
+        echo Waiting for Elasticsearch to be ready...
+        kubectl wait --for=condition=Ready pods -l app=elasticsearch --timeout=300s
+        for i in 2 3; do
+            kubectl config use-context cluster${i}
+            kubectl apply -f ./kind-e2e/logging/filebeat.yaml
+            kubectl set env daemonset/filebeat -n kube-system ELASTICSEARCH_HOST=${es_ip} ELASTICSEARCH_PORT=30000
+        done
+    fi
 }
 
 function cleanup {
-  echo "Cleanup..."
-  for i in 1 2 3; do kind delete cluster --name=cluster${i}; done
+  for i in 1 2 3; do
+
+    if [[ $(kind get clusters | grep cluster${i} | wc -l) -gt 0  ]]; then
+        kind delete cluster --name=cluster${i};
+    fi
+    done
+
```

Comment:
I'd leave some time between the deletion and cleaning up later. May be some containers need a few seconds to exit?

Reply:
From my testing it was working really good because kind deletes the containers forcefully and the second stage is more for containers that exited not related to kind.

Reply:
ahh, ok, if it does forcefull deletion, yes, it's instantaneus AFAIK. We don't need to fix what it's not broken.